### PR TITLE
AE-1019 update agent config to standardize

### PR DESCRIPTION
**Pull request by darian-ae**
This PR serves to refactor agent templates and standardize them in a config format. Most of the code refactored is to allow a user to dynamically configure multiple contract addresses per agent, whereas originally, some of the agents were written in a way to only allow for one specific address configuration.

It's been refactored in a way to use `Object.entries()` `Object.values()` and `Object.keys()` where applicable, to handle multiple addresses and contracts per configuration file. Beforehand, a lot of the agent templates were plucking data from the config directly, which does not allow for multiple contracts/addresses to work with the agent template.

### CODE REVIEW

File: new-contract-interaction/src/agent.spec.js [hunk 1]
```diff
@@ -31,8 +31,9 @@ const filteredAddress = `0x3${'0'.repeat(39)}`;
 
 const config = require('../agent-config.json');
 
+// grab first contract to test
 const [testContract] = Object.keys(config.contracts);
-const { address: testContractAddress } = config.contracts[testContract];
+const { address: testContractAddress } = config.contracts[testContract].newContractEOA;
```
Comment (Nomad-AE) [inline new-contract-interaction/src/agent.spec.js:36]: Line 35 will return an array of keys. `testContract` will not handle multiple contracts correctly. Also, `.newContractEOA` is referenced but doesn't exist in agent-config.json

File: new-contract-interaction/src/agent.spec.js [hunk 2]
```diff
@@ -72,22 +73,23 @@ describe('check agent configuration file', () => {
...
       const {
         thresholdBlockCount,
         thresholdTransactionCount,
         address,
         filteredAddresses,
         findingType,
         findingSeverity,
-      } = contracts[key];
-
+      } = contract.newContractEOA;
```
Comment (Nomad-AE) [inline new-contract-interaction/src/agent.spec.js:92]: `.newContractEOA` is referenced but doesn't exist in agent-config.json.

**Review by Nomad-AE — COMMENTED** [review]
Some minor changes recommended for overall consistency.

File: new-contract-interaction/src/agent.spec.js (hunk 2 above)
Comment (darian-ae) [reply new-contract-interaction/src/agent.spec.js:92]: oops sorry, looks like the last commit where I changed the agent-config.json file here didn't get pushed. Just pushed it up now!

File: new-contract-interaction/src/agent.spec.js (hunk 1 above)
Comment (darian-ae) [reply new-contract-interaction/src/agent.spec.js:36]: yea the reason I did this was to get an array of contracts and then just referencing the first one for testing purposes

File: new-contract-interaction/src/agent.js [hunk 3]
```diff
@@ -75,17 +75,15 @@ function provideInitialize(data) {
...
+    data.contracts = Object.entries(data.contractInfo).map(([name, entry]) => {
       const {
         thresholdBlockCount,
         thresholdTransactionCount,
         address,
         filteredAddresses,
         findingType,
         findingSeverity,
-      } = data.contractInfo[name];
+      } = entry.newContractEOA;
```
[the remark this answers is not in the document]
Comment (darian-ae) [reply new-contract-interaction/src/agent.js:86]: sorry, same as below (just pushed the updated agent config file)

File: gnosis-safe-multisig/src/agent.js [hunk 4]
```diff
@@ -27,114 +29,126 @@ function provideInitialize(data) {
...
 
-    // extract the token addresses from the Transfer events
-    data.tokenAddresses = rawLogs.map((rawLog) => rawLog.address.toLowerCase());
-
-    // get rid of any duplicates
-    data.tokenAddresses = [...new Set(data.tokenAddresses)];
+    // gnosis_safe specific configuration values
+    data.gnosisSafe = config.contracts;
+
+    data.contracts = Object.entries(data.gnosisSafe);
```
[the remark this answers is not in the document]
Comment (darian-ae) [reply gnosis-safe-multisig/src/agent.js:42]: renamed the first `data.contracts` to `safeEntries`

File: gnosis-safe-multisig/SETUP.md [hunk 5]
```diff
@@ -35,13 +35,18 @@ changes, the following content would be present in the `agent-config.json` file:
...
   "protocolName": "Synthetix",
   "protocolAbbreviation": "SYN",
-  "gnosisSafe": {
-    "address": "0xEb3107117FEAd7de89Cd14D463D340A2E6917769",
-    "version": "v1.3.0"
+  "contracts": {
+    "contractName1": {
+      "gnosisSafe": {
+        "address": "0xEb3107117FEAd7de89Cd14D463D340A2E6917769",
+        "version": "v1.3.0"
```
Comment (Nomad-AE) [inline gnosis-safe-multisig/SETUP.md:42]:
```suggestion
    "contractName1": {
      "address": "0xBC4CA0EdA7647A8aB7C2061c2E118A18a936f13D",
      "gnosisSafe": {
        "version": "v1.3.0"
```
To reflect your actual code.

**Review by Nomad-AE — COMMENTED** [review]
Looks good, I don't fully understand the gnosis agent so I didn't review that one.

File: new-contract-interaction/src/agent.spec.js (hunk 2 above)
Comment (Nomad-AE) [reply new-contract-interaction/src/agent.spec.js:92]: That makes more sense now.

File: new-contract-interaction/src/agent.spec.js (hunk 1 above)
Comment (Nomad-AE) [reply new-contract-interaction/src/agent.spec.js:36]: No problem then.

File: gnosis-safe-multisig/SETUP.md (hunk 5 above)
Comment (darian-ae) [reply gnosis-safe-multisig/SETUP.md:42]: cool cool thanks for catching all that

File: gnosis-safe-multisig/src/agent.spec.js [hunk 6]
```diff
@@ -63,24 +63,28 @@ describe('check agent configuration file', () => {
...
-    // check that there is a corresponding file for the version indicated
-    // eslint-disable-next-line import/no-dynamic-require,global-require
-    const { abi } = require(`../abi/${version}/gnosis_safe.json`);
-
-    expect(typeof (abi)).toBe('object');
-    expect(abi).not.toBe({});
+    const gnosisSafe = Object.values(config.contracts);
+    gnosisSafe.forEach((safe) => {
+      const { version } = safe.gnosisSafe;
+      const { address } = safe;
```
Comment (echoAE) [inline gnosis-safe-multisig/src/agent.spec.js:77]:
```suggestion
      const { gnosisSafe: { version }, address} = safe;
```

File: governance/src/agent.js [hunk 7]
```diff
@@ -189,20 +190,35 @@ function provideInitialize(data) {
...
-      protocolAbbreviation: agentConfig.protocolAbbreviation,
+      developerAbbreviation: config.developerAbbreviation,
+      protocolName: config.protocolName,
+      protocolAbbreviation: config.protocolAbbreviation,
     };
-    data.address = agentConfig.governance.address;
+    data.goverance = config.contracts;
+    data.contracts = Object.entries(data.goverance).map(([name, entry]) => {
+      const { abiFile } = entry.governance;
+      const { address } = entry;
```
Comment (echoAE) [inline governance/src/agent.js:200]:
```suggestion
      const { governance: { abiFile }, address } = entry.governance;
```

**Review by echoAE — APPROVED** [review]

File: governance/src/agent.js (hunk 7 above)
Comment (darian-ae) [reply governance/src/agent.js:200]: changed it to `const { governance: { abiFile }, address } = entry;`

**Pull request merged**